src/coord/coord.rs

```diff
@@ -2048,8 +2048,12 @@ fn index_sql(
     .to_string()
 }
 
+/// Testdrive uses this method to check that the in-memory state of the catalog matches
+/// the state if you were to load a new materialized instance from that catalog.
```

Comment:
Documentation like this makes me very nervous. This is the kind of thing that goes stale and gives people a false sense of security when making changes. This is a public function and can be used for any purpose! The only way to know whether your change is valid is to search for usages at the time you want to make a change.

I propose instead something like this:

```rust
Loads the catalog stored at `data_directory` and returns its serialized state.

There are no guarantees about the format of the serialized state, except that
the serialized state for two identical catalogs will compare identically.
```

Reply:
That works for me. I just need to put something in there.